boards/st/nucleo_h563zi/Kconfig.defconfig

```diff
@@ -10,6 +10,10 @@ if NETWORKING
 config NET_L2_ETHERNET
 	default y
 
+config PHY_INIT_PRIORITY
+	default 81
```

Comment:
Well, priority to 81 here is due to the fact that, as far as i have seen, DT structure drives the initialization order. 

The DT structure i chosed is that mdio is child of mac, and phy child of mdio.

Other approach would be that mdio is not a child of mac, but at same level, and to use phy-handle to get the phy, but this method didn't worked, was getting errors compiling using phy-handle = <..>in DT.

Since mdio is part of ETH IP, structure i chosed may be good. Comments are welcome.

Reply:
The way you described not working is what I did in nxp_enet , maybe look at that

Reply:
@decsny 
I see, it should work, i can eventually change in that way if you prefer, but requires some rework-retest.
If you consider this necessary, more clean, i can go that way.

Reply:
I think it would be good to be consistent within zephyr,  unless you think your current way is better

Reply:
Have you checked if the initialisation order is only required because both the phy driver and the stm32 eth mac driver (via `HAL_ETH_Init`) initialise the phy?

Reply:
@gramsay0 @decsny 

I tested same eth_nxp_enet DT structure, does not work since PHY, with defualt priorities, starts first, and i get
[00:00:00.625,000] <err> adi_adin1100_phy: PHY 0 didn't come out of reset, -116

Issue is the following:

MDIO hw is part of the ETH IP module and actually initialized by eth_stm32_hal.c only, by HAL_ETH_Init().
So PHY for this case must be initialized after ETH.